Project Management Process.md

```diff
@@ -0,0 +1,168 @@
+    Milestone A     |     Milestone 2           Goal C
+    Milestone B     |       Goal 4              Goal D
+    Milestone C     |     Milestone 3      
+                    |       Goal 5      
+  Projects          |  
+    Project 1       |   Backlog             Backlog
+    Project 2       |     Goal 1              Goal A
+    Project 3       |     Goal 2              Goal B
+                    |     Goal 3              Goal C
+                    |     Goal 4              Goal D
+                    |     Goal 5
+```
+
+### Goals
+At the heart of each project are the goals. They are the basic unit of work. Goals can be anything that ***adds value to a release***. This includes issues reported by users, feature requests, bugs, Pull-Requests, general tasks such as refactoring, documentation or research. Goals are collected, maintained and tracked in the backlog by the project lead.
+
+A goal description includes the following:
+- Name / Summary
+  - Describes what the goal is on high level
+  - Example 1: "Reduce the size of the go-ipfs executable"
+  - Example 2: "js-ipfs should have API Documentation"
+  - TODO: what is a good name for a goal?
+- Description
+  - What the problem is and how this goal solves it
+- Type
+  - Describe the type of this goal: is it a bug? a new feature? something else?
+- Status
+  - What is the current status of this goal. Is it to-be-reviewed? In progress? Blocked? Done?
+- Dependencies (where applicable)
+- Priority (optional, priority can be defined in the backlog)
+- Any additional information that helps to reach this goal (optional)
+
+TODO: a clear example
+
+### Backlog
+Backlog is the collection of all goals in a project. It should be as comprehensive as possible, including everything that needs to happen or should happen in a particular project. The backlog indicates priority by order: the items that have the highest priority are at the top of the list. The backlog is owned and maintained by the project lead. As part of daily work, the backlog gets updated and a general grooming should happen on weekly basis.
+
+A backlog has the following information:
+- All goals of a project
+- Priority of goals in sorted order
+
+While the backlog is usually one big list of goals, it may sometimes become convoluted. In this case, the backlog list can be split in two: Backlog and Icebox. The icebox section lists goals that will probably not be worked on anytime soon, or are not ready to be worked on yet, whereas the backlog itself contains the goals that are ready or "approved". If split, together they are called "the backlog" and icebox is referenced only in its specific meaning. The split and maintenance of both lists are done the project lead.
+
+### Milestones
+A milestone is a group of project's goals. Each project has a set of milestones. A milestone is completed when all goals associated with it have been completed. Milestones are tracked in the project roadmap by the project lead.
```

Comment:
can we define something more clear for what makes a milestone? maybe something along the lines of:
- A batch of goals that want to be completed around the same time.
- A batch of goals that together achieve a significant improvement to the product.
- A batch of goals that should be worked on concurrently.
- is it "a group of related issues"? i dont think so, because those could span long, long time, and we want milestones to be time oriented or at least sequentially ordered?

or maybe it's a combination of these? i'm not sure. but being clear here will help us make sure we all agree on how to "milestone" the roadmap. In fact perhaps a more actionable approach is to say:

> The Roadmap is broken down into manageable work units that combine goals into a significant improvement in the product. Milestones help make large Roadmaps achievable, and they give the team and users a clear sense of progress. Milestones may be made up of related Goals, Goals that should be completed around the same time, or Goals that together achieve a significant improvement (a larger Goal). Projects with Release schedules will likely use at least one Milestone per release. Milestones are usually, but not always, sequentially ordered, and they could be used to get a sense of time delays for features. But in practice, some Milestones may not be sequentially dependent, and the Roadmap could be rearranged.

Perhaps this, but im not sure i agree with it fully:

> There's no hard and fast way to decide what a Milestone's boundaries are. It is up to the project leader or project manager to select what Goals constitute a Milestone and what Milestones constitute a Roadmap. Sometimes it will be clear how to break down a Roadmap into manageable pieces. Sometimes it will be clear how to bundle related Goals together. The important thing is to make manageable groups (not too big) that provide a significant enough sense of progress through the Roadmap.